src/VolleyManagement.API/Controllers/AccountController.cs

```diff
@@ -0,0 +1,127 @@
+    using System.Security.Claims;
+    using System.Threading.Tasks;
+    using Model;
+    using Contracts;
+    using Contracts.Authentication;
+    using Contracts.Authentication.Models;
+    using Crosscutting.Contracts.Providers;
+
+#pragma warning disable S1200 // Classes should not be coupled to too many other classes (Single Responsibility Principle)
+    [Produces("application/json")]
+    [Route("api/[controller]/[action]")]
+    public class AccountController : Controller
+#pragma warning restore S1200 // Classes should not be coupled to too many other classes (Single Responsibility Principle)
+    {
+        private const string GOOGLE = "Google";
+        private const string COOKIE_NAME = "VMCookie";
+
+        private readonly IVolleyUserManager<UserModel> _userManager;
+        private readonly IUserService _userService;
+        private readonly ISecretsProvider _configuration;
+
+        public AccountController(
+            IVolleyUserManager<UserModel> userManager,
+            IUserService userService,
+            ISecretsProvider configuration)
+        {
+            _userManager = userManager;
+            _userService = userService;
+            _configuration = configuration;
+        }
+
+        /// <summary>
+        /// Login user
+        /// </summary>
+        /// <param name="user">json with user info provided by UI</param>
+        /// <returns>Action result</returns>
+        [HttpPost]
+        [AllowAnonymous]
+        public async Task<IActionResult> TokenSigninGoogle(string user)
+        {
+            var userInfoFromWebApp = JsonConvert.DeserializeObject<GoogleUserInfo>(user);
+
+            var validatedLoginInfoFromGoogle = await GoogleJsonWebSignature.ValidateAsync(userInfoFromWebApp.IdToken,
+                new GoogleJsonWebSignature.ValidationSettings {
+                    Audience = new List<string> { _configuration.GetGoogleClientId().Value }
+                });
+
+            var userInSystem = await _userManager.FindByEmailAsync(validatedLoginInfoFromGoogle.Email);
+
+            if (userInSystem == null)
+            {
+                userInSystem = new UserModel {
+                    Email = validatedLoginInfoFromGoogle.Email,
+                    UserName = validatedLoginInfoFromGoogle.Email,
+                    PersonName = $"{validatedLoginInfoFromGoogle.FamilyName} {validatedLoginInfoFromGoogle.GivenName}"
+                };
+                var result = await _userManager.CreateAsync(userInSystem);
+                if (!result.Succeeded)
+                {
+                    return Unauthorized();
+                }
+                else
+                {
+                    result = await _userManager.AddLoginAsync(userInSystem.Id,
+                        new UserLoginInfo(GOOGLE, userInSystem.Email));
+                    if (!result.Succeeded)
+                    {
+                        result = await _userManager.DeleteAsync(userInSystem);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable 'result'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1854)